src/Analyser/MutatingScope.php

```diff
@@ -1246,6 +1246,53 @@ private function resolveType(Expr $node): Type
 			$leftType = $this->getType($left);
 			$rightType = $this->getType($right);
 
+			if (($leftType instanceof IntegerRangeType || $leftType instanceof ConstantIntegerType) &&
+				($rightType instanceof IntegerRangeType || $rightType instanceof ConstantIntegerType) &&
+				!($node instanceof Node\Expr\BinaryOp\Pow || $node instanceof Node\Expr\AssignOp\Pow)) {
+
+				if ($leftType instanceof ConstantIntegerType) {
+					$min = $leftType->getValue();
+					$max = $leftType->getValue();
+				} else {
+					$min = $leftType->getMin();
+					$max = $leftType->getMax();
+				}
+
+				if ($rightType instanceof ConstantIntegerType) {
```

Comment:
Let's say we have `1|2|3`. I'd like to have `int<0, max> * 1|2|3` to also be resolved to `int<0, max> `.

Reply:
this union case is pretty complicated. do you think its common enough to fix it now?

Reply:
Or do you think handling `1|2|3` like `int<1 ,3>` will work for all operators?